### DISCUSSION TITLE
I set refreshInterval: 1000 at initialization, how can I cancel polling under certain conditions？

### DISCUSSION THREAD

**Discussion by Magua-Q** [vercel/swr#3025] [Q&A]
`
export const getTask = (taskId) => {
  const { data, error, mutate } = useSwr('queryAsyncResultByTaskId', () => queryAsyncResultByTaskId(taskId), { refreshInterval: 1000});
  useEffect(() => {
    if (data?.data?.status === 'SUCCESS') {
      console.log('entry', data);
      setErrorCategoryMap(data?.data)
      mutate(
        data,
        {
          revalidate,
        }
      );
    }
  }, [data, mutate]);
  return { data, error, mutate };
};
`

**Comment by suprunchuk** [comment]
```js
import useSWR from 'swr';

export const getTask = (taskId) => {
  const { data, error, mutate } = useSWR(
    'queryAsyncResultByTaskId',
    () => queryAsyncResultByTaskId(taskId),
    {
      refreshInterval: (latestData) => 
        latestData?.data?.status !== 'SUCCESS' ? 1000 : 0,
    }
  );

  useEffect(() => {
    if (data?.data?.status === 'SUCCESS') {
      console.log('entry', data);
      setErrorCategoryMap(data?.data);
      mutate(data, { revalidate: false });
    }
  }, [data, mutate]);

  return { data, error, mutate };
};

```
